File: Merge_Two_Sorted_Lists/Merge_Two_Sorted_Lists.java
class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
        next = null;
    }
}

class Solution {
    public ListNode mergedList(ListNode list1, ListNode list2) {
        ListNode head = new ListNode(0);
        ListNode current = head;

        while (list1 != null && list2 != null) {
            if (list1.val <= list2.val) {
                current.next = list1;
                list1 = list1.next;
            } else {
                current.next = list2;
                list2 = list2.next;
            }
            current = current.next;
        }

        if (list1 != null) {
            current.next = list1;
        }
        if (list2 != null) {
            current.next = list2;
        }

        return head.next;
    }
}

public class Main {
    public static ListNode toLinkedList(int[] lst) {
        ListNode head = new ListNode(lst[0]);
        ListNode current = head;
        for (int i = 1; i < lst.length; i++) {
            current.next = new ListNode(lst[i]);
            current = current.next;
        }
        return head;
    }

    public static int[] toArray(ListNode node) {
        java.util.List<Integer> result = new java.util.ArrayList<>();
        while (node != null) {
            result.add(node.val);
            node = node.next;
        }
        // Convert ArrayList to int[]
        int[] array = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            array[i] = result.get(i);
        }
        return array;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        ListNode list1 = toLinkedList(new int[]{1, 2, 3});
        ListNode list2 = toLinkedList(new int[]{2, 3, 4});

        ListNode merged = sol.mergedList(list1, list2);
        int[] result = toArray(merged);

        // Print the merged list as an array
        System.out.println(java.util.Arrays.toString(result));
    }
}

File: Merge_Two_Sorted_Lists/Merge_Two_Sorted_Lists.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def merged_list(self, list1, list2):
        head = ListNode(0)
        current = head

        while list1 and list2:
            if list1.val <= list2.val:
                current.next = list1
                list1 = list1.next

            else:
                current.next = list2
                list2 = list2.next
            current = current.next

        if list1:
            current.next = list1
        if list2:
            current.next = list2

        return head.next

def to_linked_list(lst):
    head = ListNode(lst[0])
    current = head
    for val in lst[1:]:
        current.next = ListNode(val)
        current = current.next
    return head

def to_list(node):
    result = []
    while node:
        result.append(node.val)
        node = node.next
    return result

sol = Solution()
merged = sol.merged_list(to_linked_list([1,2,3]),to_linked_list([2,3,4]))
print(to_list(merged))